Adding support for JUnit XML format (#1388)

nightwatchcyber opened the pull request:
Closes #1388

nightwatchcyber committed:
Adding support for JUnit XML format (#1388)

nightwatchcyber committed:
Adding support for JUnit XML format (#1388)

nightwatchcyber committed:
Merge remote-tracking branch 'upstream/develop' into develop

nightwatchcyber committed:
fixing changelog

nightwatchcyber committed:
fixing changelog

nightwatchcyber committed:
fixing package

nightwatchcyber committed:
fixing package

nightwatchcyber committed:
fix

nightwatchcyber committed:
fix

nightwatchcyber committed:
pipfile revert

nightwatchcyber committed:
pipfile revert

Commit:
added debugging info for metavariables into step_by_step_debugging

Commit:
pushed old version, oops, now pushing new

Commit:
remove prints

Commit:
removing unnecessary modules

nightwatchcyber committed:
added new package

nightwatchcyber committed:
added new package

nightwatchcyber committed:
trying to fix xml test

nightwatchcyber committed:
trying to fix xml test

nightwatchcyber committed:
fixing xml test

nightwatchcyber committed:
fixing xml test

nightwatchcyber committed:
fixing tox error

nightwatchcyber committed:
fixing tox error

nightwatchcyber committed:
revert

nightwatchcyber committed:
revert

Commit:
fixing attribute error

nightwatchcyber committed:
merge upstream changes

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

brendongo commented:
Thanks for the PR @nightwatchcyber! We are currently thinking through how we want to handle different output formats for semgrep going forward and will get back to early next week.

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into not_pattern_matching

Commit:
moved multiple assignment issue to Generic vs generic Assign.

Commit:
added comments

Commit:
small fixes

Commit:
small changes

Commit:
code cleanup

Commit:
run perf tests when there is a release

DrewDennison commented:
@nightwatchcyber thanks again for the PR. We've going to try to get this in this week. A couple small requests that we may get to before you do 1) can we pin `junit_xml` and `xml_diff` just to prevent against malicious updates in the future. 2) we're going to change the flag  to `--fmt json`, `--fmt sarif`, and thus `--fmt junit` just to make the API more extensible. Thanks again for the contribution!

brendongo commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 ## [Unreleased]
 - `--timeout-threshold` flag to set the maximum number of times a file can timeout before it is skipped
+- Added support for the JUnit XML report format (`--junit-xml`)
```
Should rebase and move these to a new unreleased

brendongo commented on semgrep/Pipfile:
```diff
@@ -10,9 +10,11 @@ tox = "*"
 
 [packages]
 semgrep = {editable = true,path = "."}
+junit-xml = "*"
```
No need to put this here since the dependencies in setup.py are installed when semgrep is installed
```suggestion
```

brendongo commented on semgrep/setup.py:
```diff
@@ -126,13 +126,14 @@ def run(self):
     install_requires=[
+        "attrs>=19.3.0",
         "colorama>=0.4.3",
+        "junit_xml>=1.9",
```
```suggestion
        "junit_xml==1.9",
```

brendongo commented on semgrep/tests/e2e/test_check.py:
```diff
@@ -42,6 +43,16 @@ def test_terminal_output(run_semgrep_in_tmp, snapshot):
     )
 
 
+def test_junit_xml_output(run_semgrep_in_tmp, snapshot):
```
Thanks for adding tests!

Commit:
testing

Commit:
forgot endlien

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into perf_tests

Commit:
change to bash script. If statement should work.

Commit:
edited if statement, make sure event name matches and server uses bash.

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into python_annotations

nightwatchcyber committed:
merge with upstread

nightwatchcyber committed:
fixing changelog.md

nightwatchcyber committed:
fixing changelog.md

nightwatchcyber commented:
I updated the PR to pin the file but didn't do anything about changing the CLI command yet

aryx committed:
* pfff: use latest (#1575)

aryx committed:
Do not consider .mly and .mll as ml files for semgrep (#1576)

We do not parse correctly the .mly and .mll so there's no point
considering them for semgrep.

Test plan:
pipenv shell
cd ~/pfff
semgrep -f semgrep.yml does not report anymore stuff on the .mly and .mll
inside pfff

nbrahms committed:
Fix jsonschema deprecation warning (#1566)

... by pinning validator.

mschwager committed:
Unify 'install-scripts' and 'release-scripts' to single directory (#1578)

* Unify 'install-scripts' and 'release-scripts' to single directory

* Correct for false positive

mschwager committed:
Print rule count with --validate (#1579)

mschwager committed:
Move repository-local Semgrep config to cloud app (#1577)

* Move repository-local Semgrep config to cloud app

* Update cloud config URLs

aryx committed:
* pfff: use latest (#1587)

aryx committed:
Support to search not only expr/stmts but also types (and patterns) (#1588)

When writing rules for OCaml, and when searching OCaml code, it's quite
frequent what you look for is not an expression or a statement but
a type. Moreover, we may need that soon for Typescript too where
people may want to specify rules just for the typing parts of
Typescript.

test plan:
semgrep --exclude parsing_errors -l ml -e : $X ref .

lang_ruby/parsing/parser_ruby_helpers.mli
2:val state_override : bool ref

tests/ml/visual/ref.mli
4:val y: int ref

aryx committed:
New -dump_tree_sitter_cst to dump the tree-sitter CST (#1589)

This helps to debug issues.

test plan:
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -dump_tree_sitter_cst string.js
|   | Exp_stmt
|   |   | Exp
|   |   |   Assign_exp
|   |   |   |   Choice_member_exp
|   |   |   |   | Id
|   |   |   |   | a
|   |   |   | "="
|   |   |   |   Choice_this
|   |   |   |   | Str
|   |   |   |   |   DQUOT_rep_choice_blank_DQUOT
|   |   |   |   |   | "\""
|   |   |   |   |   |   | Blank
|   |   |   |   |   | "\""
|   |   | Auto_semi
|   |   | ""

Note that it shows a bug with the Blank above instead of the content
of the actual string.

brendongo committed:
Run brew install test during release (#1586)

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into package_ellipses

Commit:
Add instructions for testing vulnerable repos to README in Examples (#1571)

brendongo committed:
Run all repo tests on pull request (#1573)

underyx committed:
CHANGELOG: Add 0.21.0's changes

underyx committed:
Release 0.21.0

brendongo committed:
Merge pull request #1593 from returntocorp/release-0.21.0

Release 0.21.0

mschwager committed:
Add true negative annotations to --test functionality (#1591)

aryx committed:
Use latest ocaml-tree-sitter-lang (#1590)

* Use latest ocaml-tree-sitter-lang

Martin fixed the issue with unnamed pattern-tokens
so we have no more Blank and have real tokens!
He also updated the Ruby grammar which was modified quite heavily
which led to quite a few changes in the CST and boilerplate.

test plan:
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -dump_tree_sitter_cst string.js
|   | Exp_stmt
|   |   | Exp
|   |   |   Assign_exp
|   |   |   |   Choice_member_exp
|   |   |   |   | Id
|   |   |   |   | a
|   |   |   | "="
|   |   |   |   Choice_this
|   |   |   |   | Str
|   |   |   |   |   DQUOT_rep_choice_imm_tok_pat_de5d470_DQUOT
|   |   |   |   |   | "\""
|   |   |   |   |   |   | Imm_tok_pat_de5d470
|   |   |   |   |   |   | foo
|   |   |   |   |   | "\""
|   |   | Auto_semi
|   |   | ""

We now have the content of the string 'foo'! no more Blank

* * semgrep-core/ocaml-tree-sitter-lang: more latest

* * semgrep-core/ocaml-tree-sitter-lang: use latest!

* * semgrep-core/parsing/Parse_javascript_tree_sitter.ml: fix unused var warning

* new -dump_ast_pfff cli action

aryx committed:
* pfff: use latest (#1599)

Commit:
Add a 'none' language to config files to support arbitrary regex scanning. (#1584)

* Add a 'none' language to config files.

'none' language only supports 'pattern-regex', 'patterns', and 'pattern-either' clauses.
The 'none' language makes it easy to support scanning using pattern-regex without having
to specify target files on the CLI.

This also helps address the divergent behavior between includes on the command line and
the 'paths' and 'include' key. The 'paths' and 'include' key did not explicitly scan
included files when in the config file. This behavior persists UNLESS using the 'none'
language, in which case it will scan all files (still respecting the behavior of
all includes and excludes, both on the CLI and the config file.

This commit also added support for HTML comments with '--test'.

Tests are included for this feature.

* Revert __init__.py

* Try to reset pfff/ to develop

* Fix broken span when validating none-language rules

* Make NONE_LANGUAGE the Language type

* Fix issue where I accidentally cut and paste lines in the wrong place -_-

* Update CHANGELOG.md

aryx committed:
Move pfff under semgrep-core (#1601)

* Move pfff under semgrep-core

This is part1 of using pfff with dune

test plan:
(under semgrep-core)
make
make test

* update install procedure for pfff

* * Dockerfile: typo

* * semgrep-core/pfff: use latest

* use latest again

mschwager committed:
Remove unused variable (#1604)

Commit:
Merge branch 'develop' of github.com:returntocorp/semgrep into package_ellipses

Commit:
adding pfff changes

Commit:
oops

Commit:
changing the types to match changed parser_java.mly

Commit:
adding tests

aryx committed:
Switch to tree-sitter first for parsing Javascript (#1603)

* Switch to tree-sitter first for parsing Javascript

test plan:
make
make test

* * semgrep-core/parsing/Parse_javascript_tree_sitter.ml: fix traduction of
new A(1) to match what we do in pfff

* more fixes

test plan:
(semgrep) pad@yrax:~/github/semgrep$ pytest -k test_nosem_rule semgrep/tests/e2e/test_check.py -vv

now pass!

* Remove the default timeout for Javascript

This should fix https://github.com/returntocorp/semgrep/issues/1471

Now that we use tree-sitter, no need to use a timeout of 5s for Javascript
parsing (was especially useful on minified files). Tree-sitter handles
well minified-files.

test plan:
make test

* * semgrep/tests/public_repos.py: back to xfail

aryx committed:
Merge branch 'develop' into ellipsis_annotations

aryx committed:
Merge branch 'ellipsis_annotations' into develop

aryx committed:
Use PCRE regexp syntax in semgrep "=~/xxx/" pattern (#1610)

* Support PCRE-compatible regexps in semgrep "=~/xxx/" pattern

This fixes
https://github.com/returntocorp/semgrep/issues/1514
and
https://github.com/returntocorp/semgrep/issues/688

test plan:
test files included

* * semgrep-core/matching/Matching_generic.ml: comment

underyx committed:
Rename packs to rulesets in the repo (#1612)

mschwager committed:
Parallelize --test functionality (#1611)

* Parallelize --test functionality

* Explicitly state why we're avoiding multiprocessing when testing

mschwager committed:
Remove unnecessary indirection (#1616)

mschwager committed:
Remove install-script release functionality (#1615)

* Remove install-script release functionality

* Bump CHANGELOG

* Remove additional unnecessary behavior

aryx committed:
Using new bracket in ArrAccess (#1617)

This should fix
https://github.com/returntocorp/semgrep/issues/1285

test plan:
make test

mschwager committed:
Fail Semgrep rules tests on exit != 0 (#1618)

aryx committed:
* CHANGELOG.md: update (#1620)

Commit:
Release 0.22.0

brendongo committed:
Merge pull request #1621 from returntocorp/release-0.22.0

Release 0.22.0

colleend committed:
tests for package ellipsis bug (#1622)

* tests for go package fix along with necessary changes in pfff go parser

* renamed files

* adding pfff changes

* tests for package ellipsis bug

* merging

* what

Co-authored-by: Colleen Dai <chai@Colleens-MBP.attlocal.net>

aryx committed:
* semgrep-core/pfff: use latest (#1624)

aryx committed:
Use latest pfff and new [@@profiling] attribute (#1625)

test plan:
you can see the Apply_equivalences.apply below when running with -profile

d@yrax:~/semgrep$ yy -lang js -profile -e 'FOO' tests/js/concrete_syntax.js
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -lang js -profile -e FOO tests/js/concrete_syntax.js
Profile mode On
disabling -j when in profiling mode
---------------------
profiling result
---------------------
Main total                               :      0.010 sec          1 count
Parallel.invoke                          :      0.000 sec          1 count
Common.=~                                :      0.000 sec         26 count
Semgrep.check                            :      0.000 sec          1 count
Semgrep.match_e_e                        :      0.000 sec         16 count
file_type_of_file                        :      0.000 sec          1 count
rule:-e/-f                               :      0.000 sec         16 count
Parse_js.tokens                          :      0.000 sec          1 count
Naming_ast.resolve                       :      0.000 sec          1 count
Constant_propagation.xxx                 :      0.000 sec          1 count
Common.full_charpos_to_pos_large         :      0.000 sec          1 count
Apply_equivalences.apply                 :      0.000 sec          1 count
Unix.stat                                :      0.000 sec          2 count

mschwager committed:
Add ability to read target contents from stdin (#1628)

aryx committed:
Use the new [@@profiling] ppx extension (#1631)

This removes some boilerplate code that can be autogenerated
by the ppx_profiling extension in pfff/ppx.

test plan:
You can see Semgrep_generic.match_e_e because of the
[@@profiling] attribute.

/home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -lang js -profile -e FOO tests/js/concrete_syntax.js
Profile mode On
disabling -j when in profiling mode
---------------------
profiling result
---------------------
Main total                               :      0.006 sec          1 count
Parallel.invoke                          :      0.001 sec          1 count
Semgrep.check                            :      0.000 sec          1 count
Common.=~                                :      0.000 sec         26 count
Semgrep_generic.match_e_e                :      0.000 sec         16 count
rule:-e/-f                               :      0.000 sec         16 count
file_type_of_file                        :      0.000 sec          1 count
Parse_js.tokens                          :      0.000 sec          1 count
Naming_ast.resolve                       :      0.000 sec          1 count
Constant_propagation.xxx                 :      0.000 sec          1 count
Common.full_charpos_to_pos_large         :      0.000 sec          1 count
Apply_equivalences.apply                 :      0.000 sec          1 count
Unix.stat                                :      0.000 sec          2 count

brendongo committed:
ci: allow running on demand (#1635)

aryx committed:
New semgrep_core -diff_pfff_tree_sitter CLI action (#1640)

This can help find bugs on differences between what is parsed
by pfff and tree-sitter. This is useful because right now we
use pfff to parse the pattern, and sometimes tree-sitter to parse the code,
and if they disagree on how they parse things, then we may not find matches.

test plan:
```
diff_tsitter_pfff/home/pad/semgrep/tests/PARSING $ yy -diff_pfff_tree_sitter string.go
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -diff_pfff_tree_sitter string.go
NOTE: consider using -full_token_info to get also diff on tokens
diff_tsitter_pfff/home/pad/semgrep/tests/PARSING $ yy -full_token_info -diff_pfff_tree_sitter string.go
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -full_token_info -diff_pfff_tree_sitter string.go
NOTE: consider using -full_token_info to get also diff on tokens
--- /tmp/tmp-16886-dce4d9.x	2020-09-04 10:08:31.678766347 +0200
+++ /tmp/tmp-16886-293f78.x	2020-09-04 10:08:31.678766347 +0200
@@ -46,9 +46,8 @@
                                ("foo",
                                 { token =
                                   (OriginTok
-                                     { str = "\"foo\""; charpos = 37;
-                                       line = 3; column = 12;
-                                       file = "string.go" });
+                                     { str = "\""; charpos = 37; line = 3;
+                                       column = 12; file = "string.go" });
                                   transfo = NoTransfo }))))
                    ))
                  ],
```

aryx committed:
Fix Parse_tree_sitter_helpers.combine_infos (#1641)

This will help https://github.com/returntocorp/semgrep/issues/1609

We were not generating the right Parse_info.t (aka token info) in
tree-sitter when we had to combine multiple tokens in one.
Indeed, in pfff we currently parse certain strings as a single
token, but in tree-sitter this was decomposed in 3 tokens, one for the
opening ", one for the content, and one for the closing ".
Before this diff, we were generating the right literal content, but
the token associated with it was just the token for the opening ".
This led to some issues in the editor when a message was mentioning
a metavariable that was bound to a string, because we are using the
tokens to pretty print its content.

test plan:
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -full_token_info -diff_pfff_tree_sitter string.go
NOTE: consider using -full_token_info to get also diff on tokens

No more diff!

mschwager committed:
Remove Semgrep binary checks from ubuntu release (#1627)

underyx committed:
Fix typo in CHANGELOG (#1623)

mschwager committed:
Separate data generation and text output in --test (#1633)

nbrahms committed:
Improve jsonschema error when missing top-level key (#1642)

Fixes #1619.

ievans committed:
report # of files ran on (#1643)

Co-authored-by: Isaac Evans <ine@mit.edu>

clintgibler committed:
Update CLI help messages with path to appropriate docs (#1646)

Update CLI help messages with paths to the appropriate docs.

🙏 Erik Cabetas for the feedback!

mjambon committed:
minimal typescript support (#1634)

* Add .ocp-indent config, for "standard" ocaml indentation, if you use
ocp-indent for indenting code automatically (e.g. when you press TAB in
emacs).

* Add translation of tree-sitter typescript CST to javascript AST (WIP)

* Revert the Parse_javascript_tree_sitter_priv rename to avoid conflicts.
Remove the mli instead.

* WIP (at "Abst_class_decl")

* WIP

* WIP

* Finished initial translation of the tree-sitter typescript boilerplate
into javascript AST.

* Use tree-sitter parser for typescript

* Fix type error

* Some debug logging

* Fix type errors introduced by new 'combine_infos'

* Add the prefix 'todo_' to each function that should not be called.

* Run all the javascript tests as typescript tests except for jsx, since
we don't support tsx yet.

* Add just one test case specific to typescript.

Co-authored-by: Martin Jambon <Martin Jambon>

IagoAbal committed:
tests: java: Add regression test for #877 (#1637)

aryx committed:
* semgrep-core/pfff: use latest and CEs on Parse_ruby_tree_sitter.ml (#1648)

aryx committed:
Use new let* monadic let to replace >>= in Generic_vs_generic (#1649)

Using cutting edge OCaml!

test plan:
make

aryx committed:
Use latest pfff with improved Ruby strings representation (#1650)

* Use latest pfff with improved Ruby strings representation

This should help
https://github.com/returntocorp/semgrep/issues/1609

test plan:
make
make test

* * semgrep-core/parsing/Parse_ruby_tree_sitter.ml: use latest pfff

* more

* * semgrep-core/pfff: latest

aryx committed:
syncweb (#1651)

* * semgrep-core/matching/Generic_vs_generic.ml: syncweb

* syncweb

aryx committed:
* semgrep-core/pfff: use develop branch (#1653)

aryx committed:
Delete Typescript  symbolic links test files (#1655)

* Delete Typescript  symbolic links test files

I think it is simpler to add a test to run the typescript
parser on the js/ dir

test plan:
make test

* * semgrep-core/parsing/Test_parsing.ml: support -lang typescript
in -dump_tree_sitter_cst

aryx committed:
Using latest pfff with types for type constructs in ast_js.ml (#1656)

* Using latest pfff with types for type constructs in ast_js.ml

This is for https://github.com/returntocorp/semgrep/issues/1632

Test plan:
make test

* Adding types for typescript variables in JS AST

test plan:
test files included!

aryx committed:
Fix timeout handling on languages using tree-sitter and pfff (#1657)

This also switches to float for specifying timeout (easier to test things).

We were incorrectly capturing the Timeout exn in Parse_code
when the first parser (usually the tree-sitter one) was timing out.
This then caused to use the second parser (usually the pfff one) without
any more timeout.

test plan:
  semgrep -l js --timeout 0.5 -e 'FOO' tests/PERF/timeout.js
now finishes quickly and does not invoke the pfff parser.

aryx committed:
* semgrep-core/pfff: use latest (#1658)

aryx committed:
* semgrep-core/semgrep.yml: Starting dogfooding (#1659)

* * semgrep-core/semgrep.yml: Starting dogfooding

Dogfooding!

test plan:
wait for CI to kick in and expect no failure

* * .circleci/config.yml: misc

* * semgrep-core/bin/Main.ml: test

* * semgrep-core/bin/Main.ml: more

* * semgrep-core/bin/Main.ml: remove errors

sabrinabrogren committed:
updating version to 23

colleend committed:
Js ellipsis object (#1660)

* tests for go package fix along with necessary changes in pfff go parser

* renamed files

* adding pfff changes

* adding tests for js ellipsis parsing

* pfff

Co-authored-by: Colleen Dai <chai@Colleens-MBP.attlocal.net>

aryx committed:
Misc developer rules in Makefile and added old test files (#1663)

* Misc developer rules in Makefile and added old test files

test plan:
make test

* * semgrep-core/Makefile: commit rules

aryx committed:
Handling more types in typescript (#1664)

* * semgrep-core/Makefile: misc

* * semgrep-core/parsing/Parse_javascript_tree_sitter.ml: switching to the new
field_classic record type in ast_js.ml

* Handling types in parameters in typescript

test plan:
test file included

* Handling types in fields in Typescript

test plan:
test file included
make test

DrewDennison committed:
Update config.yml

DrewDennison committed:
Update config.yml

DrewDennison committed:
Update config.yml

brendongo committed:
Merge pull request #1661 from returntocorp/release-0.23.0

Release 0.23.0

brendongo committed:
ci: performance first runs baseline to account for hardware differences (#1513)

brendongo committed:
pypi: publish manylinux wheel as any so works on alpine (#1638)

aryx committed:
Use latest pfff (no more json-wheel deps, no perl dep, etc.) (#1669)

test plan:
make
make test

brendongo committed:
ci: autobump homebrew core on release tag (#1667)

brendongo committed:
ci: upload to pypi on release (#1662)

aryx committed:
Use latest pfff and remove deps to pfff in opam (#1671)

mjambon committed:
Use base image for docker build (#1668)

* Use ocaml base images hosted on r2c's DockerHub account.

* Use base ocaml image with pre-installed opam packages for
the dockerfile build.

* Use immutable image for main dockerfile

Co-authored-by: Martin Jambon <Martin Jambon>

aryx committed:
* Dockerfile: add git in semgrep docker image (#1672)

This seems necessary for circleCI to be able to apply semgrep in CI
on git repositories using submodules.
See https://app.circleci.com/pipelines/github/returntocorp/semgrep/785/workflows/8bc50c5e-7891-45e6-8111-1ad8224ab79b/jobs/1418 for an example of error

test plan:
wait for it to be accepted, pushed to docker, and check if next
PR get a working circle CI

aryx committed:
Use latest pfff (#1676)

This should fix the error regarding ounit2 and JSON in CI

aryx committed:
* semgrep-core/pfff: use latest (#1677)

* * semgrep-core/pfff: use latest

* * Dockerfile: adding ssh too

* * Dockerfile: openssh

aryx committed:
* semgrep-core/tests/ocaml/dots_params.sgrep: new test file (#1678)

aryx committed:
[Dogfood] New semgrep rule to prevent printing on stdout on semgrep codebase (#1679)

This can help https://github.com/returntocorp/semgrep/issues/1630

test plan:
make check

DrewDennison committed:
enable typescript extentions (#1680)

aryx committed:
Add basic support for PHP in semgrep (#1681)

This will help https://github.com/returntocorp/semgrep/issues/1143

test plan:
test files included

DrewDennison committed:
add in php ext (#1682)

mschwager committed:
Respect --json flag with --test (#1665)

* Respect --json flag with --test

Fix #1592.

* Add CHANGELOG note

* Address Ajin's feedback

mjambon committed:
Add TSX support (#1683)

* Use latest ocaml-tree-sitter-lang, which now includes tsx.

* Add TSX support from tree-sitter.

Co-authored-by: Martin Jambon <Martin Jambon>

Commit:
Changelog for 0.24.0

Commit:
Bump version, add `make bump`

nbrahms committed:
Update CHANGELOG.md

brendongo committed:
Merge pull request #1684 from returntocorp/release-0.24.0

Release 0.24.0

mjambon committed:
Guess typescript dialect (ts or tsx) from file extension, unless forced. (#1686)

Co-authored-by: Martin Jambon <Martin Jambon>

mschwager committed:
Add env variable for appending to user agent (#1685)

aryx committed:
Fixing todos in Parse_typescript_tree_sitter.ml (#1687)

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: fixing some todo ()

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: progress

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml:

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/pfff: use latest

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: merge property_prop
and fun_prop

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: progress

* * semgrep-core/parsing/Parse_javascript_tree_sitter.ml: misc

* CEs

* CEs on decorators

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: progress

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_javascript_tree_sitter.ml: more

* * semgrep-core/parsing/Parse_typescript_tree_sitter.ml: Final todo!

aryx committed:
Unspecified type in let matches let with type (#1698)

Another "unespecified matches more" principle case.
We were doing it for function definition signature, and parameter,
but we forgot to do for variable definitions.

test plan:
unit test files included
make test

aryx committed:
Test file for deep expression visitor operator in C (#1706)

This closes https://github.com/returntocorp/semgrep/issues/1694

test plan:
test file included
make test

aryx committed:
Use new python rule to detect wrong string concatenation in sets (#1707)

* Use new python rule to detect wrong string concatenation in sets

This should help detect bugs like the one reported in
 https://github.com/returntocorp/semgrep/issues/1705

test plan:
make check
wait for CI to see if circleCI spot the reported bug above.

* BUGFIX!

aryx committed:
Test file for arrow pattern matching arrows stored in vars (#1710)

Arrows in vars used to be consider toplevel functions, when really
they should still be considered lambdas.

test plan:
test files included
make test

srenatus committed:
docs/pattern-features.md: fix golang snippet (#1708)

I'm not 100% certain that this was the original intention behind the code
snippet, but it seems probable to me. The code that was there previously wasn't
valid.

Signed-off-by: Stephan Renatus <srenatus@chef.io>

aryx committed:
Test files for empty yield (#1711)

This closes https://github.com/returntocorp/semgrep/issues/1688

test plan:
test file included
make test

IagoAbal committed:
tests: python: Add regression test for pfff#289 (#1690)

Follows: pfff@0ae0779e6 ("generic: Fix bug in constant propagation for Python (#289)")

aryx committed:
Test file for '...' between statements for PHP (#1716)

This closes https://github.com/returntocorp/semgrep/issues/1715

test plan:
test file included
make test

mschwager committed:
Use stdin target instead of temp files (#1696)

* Use stdin target instead of temp files

* Remove unnecessary shellcheck exemption

* Update validate-docker-build reference

Commit:
Regenerate pipfile

Commit:
Address PR comments: remove junit-xml from Pipefile and pin junit-xml version in setup.py

Commit:
Regenerate pipfile

Commit:
Fix syntax error in tests after merge

Commit:
Merge

Commit:
Regen Pipfile.lock and fix CHANGELOG CLI flag

brendongo commented:
Merged in #1720. Thanks @nightwatchcyber!
